script.js

```diff
@@ -0,0 +1,230 @@
+const input = document.getElementById('input');
+const reposContainer = document.getElementById('repos__container');
+const sortDetails = document.getElementById('sort__details');
+const sortBtns = document.querySelectorAll('.sort__btn');
+const sortStates = {
+    stars: 0,
+    popular: 1,
+    original: 2
+};
+let curInputVal = '';
+let repoList = []; // Total repos
+let displayedRepoList = []; // Displaying repos
+let curSortCriteria = 'stars';
+
+
+sortBtns.forEach(function(button) {
+    button.addEventListener('click', function(){
+        const value = button.getAttribute('data-value');
+        setSortCriteria(value);
+    })
+})
+
+
+function setSortCriteria(criteria) {
+    sortBtns.forEach(function(button) {
+        button.classList.remove('sort-button--active');
+    })
+    sortBtns[sortStates[criteria]].classList.add('sort-button--active');
+
+    if(criteria != curSortCriteria) {
+        curSortCriteria = criteria;
+        let curSort = curSortCriteria[0].toUpperCase() + curSortCriteria.slice(1);
+        let sortInfo = `
+            <b>${curSort}</b>
+        `;
+        sortDetails.innerHTML = 'Current sort criteria: ';
+        sortDetails.insertAdjacentHTML('beforeEnd', sortInfo);
+        
+        reposContainer.innerHTML = '';
+        showRepos();
+    }
+}
+
+
+/* Event delegation to remove repo card */
+reposContainer.onclick = (event) => {
+    if(event.target.className == 'del__button') {
+        let closestEle = event.target.closest('.repo__card');
+
+        closestEle.style.maxHeight = '0';
+        closestEle.style.margin = '0';
+        setTimeout(() => {
+            closestEle.remove();
+        }, 500);
+    }
+}
+
+
+/* Process input data on Enter */
+input.addEventListener('keypress', function(event) {
+    if(event.key == 'Enter'){
+        getInputVal();
+    }
+});
+
+
+function getInputVal() {
+    if(input.value != '' && input.value != curInputVal){
+        // console.log(input.value);
+        repoList = [];
+        curInputVal = input.value;
+        handleInput(input.value);
+    }
+}
+
+
+async function handleInput(name) {
+    let li = name.split(' ').join('');
+    const nameList = li.split(',');
```

Comment:
Should properly handle empty and duplicate input:

![image](https://user-images.githubusercontent.com/12707960/172046387-ed98a803-5be6-4489-82b6-528840a4f7da.png)

![image](https://user-images.githubusercontent.com/12707960/172046423-fcb92401-cc84-4bc9-be4f-3d2afb9950a6.png)